### PR TITLE
Maintenance via token storage

### LINKED ISSUE #12: Go past maintenance page with a cookie instead of an IP
Restricting based on IPs is not always feasible. Eg. mobile 4G connections have dynamic IPs, corporate firewalls might pose problems, things can go south pretty fast.

in the past 4-5 years, the way I tackled going beyond maintenance page was with secret cookies:
* you have the cookie, you see the website
* you don't have the cookie, you see the maintenance page

you can get the cookie by manually fiddling around with your browser.
or by accessing a secret url that creates that cookie in your browser.
or through a bookmarklet. 

this way, you can go past the maintenance page by using any kind of devices.

this came off from a Twitter conversation: https://twitter.com/oallain33/status/1449299407741067266

### PR DESCRIPTION
**Pull request by ghost**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?   | no
| Fixed issue | #12

### CODE REVIEW

File: src/Resources/config/admin_routing.yaml [hunk 1]
```diff
@@ -7,3 +7,9 @@ sylius_admin_maintenance_configuration:
             permission: true
             section: admin
             alias: plugin_synolia_maintenance
+
+sylius_admin_token_storage:
+    path: //maintenance/token-storage
+    methods: [GET, POST]
+    defaults:
+        _controller: Synolia\SyliusMaintenancePlugin\Controller\TokenStorageController
```
Comment (oallain) [inline src/Resources/config/admin_routing.yaml]:
```suggestion
sylius_admin_token_storage:
    path: /maintenance/token-storage
    methods: [GET, POST]
    defaults:
        _controller: Synolia\SyliusMaintenancePlugin\Controller\TokenStorageController
        _sylius:
            permission: true
            section: admin
            alias: plugin_synolia_maintenance
```

Double / in path.
IMO, we need to check permission too.

File: src/Resources/translations/messages.fr.yml [hunk 2]
```diff
@@ -6,6 +6,10 @@ maintenance:
         form:
             enabled: Activé ?
             ip: Adresses Ip (séparées par une virgule, sans espace)
+            token_storage:
+                button: Générer token
+                help: Autoriser l'accès à votre site d'internet en utilisant un token secret
```
Comment (oallain) [inline src/Resources/translations/messages.fr.yml]:
```suggestion
                button: Générer un token
                help: Autoriser l'accès à votre site internet en utilisant un token secret
```

**Review by oallain — DISMISSED** [review]

File: src/Checker/TokenChecker.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+use Synolia\SyliusMaintenancePlugin\Voter\IsMaintenanceVoterInterface;
+
+class TokenChecker implements IsMaintenanceCheckerInterface
+{
+    private TokenStorage $storage;
+
+    public function __construct(TokenStorage $storage)
+    {
+        $this->storage = $storage;
+    }
```
Comment (flochntrl) [inline src/Checker/TokenChecker.php:19]: Tu peux simplifier la ligne 14 à 19 : 

```
   public function __construct(private TokenStorage $storage)
    {
    }
```

Comment (flochntrl) [reply src/Checker/TokenChecker.php:19]: Ok je viens de voir la compatibilité 7.4

**Review by flochntrl — DISMISSED** [review]

File: src/Checker/TokenChecker.php (hunk 3 above)
Comment (oallain) [reply src/Checker/TokenChecker.php:19]: Yes ;)

**Comment by oallain** [comment]
Oops, I merged in the wrong order.

**Pull request merged**